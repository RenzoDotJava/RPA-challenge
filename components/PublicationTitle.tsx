import { View, Text, Platform, StyleSheet } from "react-native";

interface PublicationTitleProps {
	userId?: number;
}

const PublicationTitle = ({ userId }: PublicationTitleProps) => {
	return (
		<View style={styles.container}>
			<Text style={styles.title}>User {userId}</Text>
			<View style={styles.line}></View>
		</View>
	);
};

const styles = StyleSheet.create({
	container: {
		backgroundColor: "#DFDFDF",
		marginTop: 10,
		paddingHorizontal: 13,
	},
	title: {
		marginTop: 15,
		marginBottom: 8,
		color: "#32373A",
		fontWeight: "bold",
		fontSize: 25,
		fontFamily: Platform.OS === "android" ? "serif" : "Arial",
	},
	line: {
		backgroundColor: "#32373A",
		height: 4,
		width: 45,
	},
});

export default PublicationTitle;
