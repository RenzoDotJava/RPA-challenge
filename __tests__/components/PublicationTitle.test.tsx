import { render, screen } from "@testing-library/react-native";
import PublicationTitle from "../../components/PublicationTitle";

const props = {
	userId: 1,
};

describe("Testing <PublicationTitle />", () => {
	it("Matching snapshot", () => {
		render(<PublicationTitle {...props} />);

		expect(screen).toMatchSnapshot();
	});

	it("Showing the props", () => {
		render(<PublicationTitle {...props} />);

		expect(screen.getByText(`User ${props.userId}`)).toBeTruthy();
	});
});
