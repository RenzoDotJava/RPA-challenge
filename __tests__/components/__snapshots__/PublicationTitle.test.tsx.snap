// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <PublicationTitle /> Matching snapshot 1`] = `
<View
  style={
    {
      "backgroundColor": "#DFDFDF",
      "marginTop": 10,
      "paddingHorizontal": 13,
    }
  }
  testID="publication-title"
>
  <Text
    style={
      {
        "color": "#32373A",
        "fontFamily": "Arial",
        "fontSize": 25,
        "fontWeight": "bold",
        "marginBottom": 8,
        "marginTop": 15,
      }
    }
  >
    User 
    1
  </Text>
  <View
    style={
      {
        "backgroundColor": "#32373A",
        "height": 4,
        "width": 45,
      }
    }
  />
</View>
`;
